src/identity/ed25519/ed25519-key-pair-identity.ts

```diff
@@ -30,7 +30,15 @@ export class Ed25519KeyPairIdentity extends PublicKeyIdentity {
       throw new Error("Invalid Mnemonic")
     }
     const seed = bip39.mnemonicToSeedSync(sanitized).slice(0, 32)
-    const keys = ed25519.generateKeyPair({ seed })
+    return Ed25519KeyPairIdentity.fromSeed(seed)
+  }
+
+  static fromSeed(seed: string | pki.ed25519.NativeBuffer) {
+    let _seed = seed?.slice(0, 32)
+    if (_seed.length !== 32) {
```

Comment:
If we're already slicing then we're really only testing if it's _less_ than 32 characters, right?